ts/util/table-builder.ts

```diff
@@ -271,7 +277,31 @@ export class TableBuilder<ParentType, ChildType> {
           );
           break;
         case "text":
```

Comment:
`addFilterControls` has gotten pretty long. Maybe separate the cases out into their own functions? e.g.

```
function addFilterControls(...) {
  ...
  switch (filterType) {
    case "dropdown":
      addDropdownFilter(...);
      return;
    case "text":
      addTextFilter(...);
      return;
    default:
      ...
  }
}
```